### DISCUSSION TITLE
How do you include a git submodule in a Dagster Cloud serverless deployment?

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#17823] [Q&A]
What''s the best way to install a private python module hosted on Github inside Dagster Cloud?

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C03SMUY0SFQ/p1699462979161219?thread_ts=1699462979.161219&cid=C03SMUY0SFQ).*

**Comment by CharlesFr** [comment]
To be clear, the question is regarding adding a private submodule to Dagster Cloud, which is initialised like this:

`git submodule add <repository-url> <destination-path>
`

**Reply by garethbrickman** [reply]
When using a Hybrid deployment of Dagster+ your code must be packaged into a Docker image and pushed to a registry your agent can access (­[reference doc](https://docs.dagster.io/dagster-plus/managing-deployments/code-locations#dagster-code-requirements)). So as part of the CI/CD process you use to do that you could clone the necessary repositories or submodules into your Dagster project.
